packages/flutter/lib/src/rendering/object.dart

```diff
@@ -819,6 +834,146 @@ class PipelineOwner {
+  /// [RenderPhysicalShape] with regard to painting order. This number is not
+  /// the same as the number of total [RenderObject]s in the tree, but rather the
+  /// total number of [RenderPhysicalModel] or [RenderPhysicalShape]s to check.
+  ///
+  /// Setting this number too high will degrade performance if many render
+  /// objects using elevation are present in the tree all are used correctly.
+  /// See [debugCheckElevationData] for a discussion of the performance
+  /// characterstics of the check.
+  ///
+  /// This property is only used when asserts are enabled (e.g. debug mode).
+  ///
+  /// The default value is 10.
+  ///
+  /// This value must be >= 0 and must not be null. Setting it to 0 will
+  /// disable the check, and a warning will be printed once if there are more
+  /// objects available than actually being checked.
+  set maxElevationObjectsToCheck(int value) {
+    assert(value != null);
+    assert(value >= 0);
+    _maxElevationObjectsToCheck = value;
+  }
+
+  Map<RenderObject, List<_ElevationData>> _elevations;
+  Path _translatePath(Path p, Offset offset) {
+    final Matrix4 matrix = Matrix4.identity()..translate(offset.dx, offset.dy);
+    return p.transform(matrix.storage);
+  }
+  bool _printedExceededmaxElevationObjectsToCheckWarning = false;
+
+  /// Checks for elevation consistency with regard to painting order.
+  ///
+  /// This method is only meant for debug builds, and should only be called from
+  /// an `assert` from within the framework. All of the parameters must not be
+  /// null.
+  ///
+  /// The [elevation] refers to the elevation specified on the
+  /// [RenderPhysicalModel] or [RenderPhysicalShape]. The [area] parameter
+  /// specifies the [Rect] on the screen in which the physical model will be
+  /// drawn, and the [path] parameter specifies the actual [Path] to be drawn.
+  /// The path is expected to be smaller than the area.
+  ///
+  /// If the durrent paint cycle results in drawing a shape at an elevation
+  /// lower than another shape already drawn in this area, a [Path] will be
+  /// returned describing the boundary of the conflicting shapes. Callers can
+  /// use this path to highlight the area of the screen where the conflict
+  /// occurs.
+  ///
+  /// This check has worst case performance of O(Σn) time, which is better than
+  /// O(n^2) but worse than O(n log(n)) (its curve is similar to O(n^1.8)). If
+  /// the check degrades performance unacceptably for debug usage, consider
+  /// setting [maxElevationObjectsToCheck] to a lower number, or to 0 to disable
+  /// it completely. The number can still be set higher in unit tests, where
+  /// performance is less critical.
+  Path debugCheckElevationData(
+    double elevation,
+    Rect area,
+    Path path,
+    RenderObject object,
+    RenderObject lastCommonAncestor,
+  ) {
+    assert(elevation != null);
+    assert(area != null);
+    assert(path != null);
+    assert(object != null);
```

Comment:
assert that lastCommonAncestor is not null? (or update documentation that it may be null explaining what that would mean)?

Reply:
Done.